packages/xforms-engine/src/lib/reactivity/createAssertedMemo.ts

```diff
@@ -0,0 +1,18 @@
+import { createMemo, type Accessor } from 'solid-js';
+import type { ReactiveScope } from './scope.ts';
+
+export const createAssertedMemo = <T, U extends T>(
+	scope: ReactiveScope,
+	fn: Accessor<T>,
+	assert: (value: T) => asserts value is U
+): Accessor<U> => {
+	return scope.runTask(() => {
+		return createMemo((): U => {
+			const value = fn();
+
+			assert(value);
+
+			return value;
+		});
+	});
+};
```

Comment:
looks like unused code

Reply:
Yeah, this and `assertions.ts` were part of a naive first pass, but didn't turn out to be needed. A little bit of how the sausage wasn't made 🙃.

As mentioned in the commit removing them, I think it might be worth keeping the original commit intact for slightly easier future lookup, because they're general enough we might find them useful for something else.